test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,182 @@
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 8015854
+ * @summary JButton text set as html image had additional unwanted padding
+ * @run main HtmlButtonImageTest
+ */
+
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Insets;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.awt.Robot;
+import java.net.URL;
+import java.nio.file.Path;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.SwingConstants;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.imageio.ImageIO;
+import java.io.IOException;
+import java.io.File;
+
+
+public final class HtmlButtonImageTest {
+    private static JFrame frame;
+    private static Point point;
+    private static URL urlImage;
+    private static JButton button;
+
+    private static Path srcDir;
+
+    public static final int BUTTON_HEIGHT = 37;
+    public static final int BUTTON_WIDTH = 37;
+    public static final int SQUARE_HEIGHT = 19;
+    public static final int SQUARE_WIDTH = 19;
+    public static final int PIXEL_BUFFER = 3;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoDelay(2000);
+        robot.setAutoWaitForIdle(true);
+
+        // store path to source directory to locate image
+        srcDir = Path.of(System.getProperty("test.src", "."));
+
+        // generate red_square.png image to use as JButton text
+        SwingUtilities.invokeAndWait(() -> {
+            generateImage();
+        });
+
+        // cycle test through all available LAFs
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            SwingUtilities.invokeAndWait(() -> {
+                setLookAndFeel(laf);
+                createAndShowGUI();
+            });
+
+            // retrieve color of pixels at each edge of square image by starting at the center of the button
+            robot.mouseMove(frame.getLocationOnScreen().x, frame.getLocationOnScreen().y);
+            robot.mouseMove(button.getLocationOnScreen().x, button.getLocationOnScreen().y);
+
+            setupCenterCoord();
+            robot.mouseMove(point.x, point.y);
+
+            // store each pixel color on the edge of each side of the red square
+            Color leftClr = robot.getPixelColor(point.x - (SQUARE_WIDTH/2) + PIXEL_BUFFER, point.y);
+            Color rightClr = robot.getPixelColor(point.x + (SQUARE_WIDTH/2) - PIXEL_BUFFER, point.y);
+            Color topClr = robot.getPixelColor(point.x, point.y - (SQUARE_HEIGHT/2) + PIXEL_BUFFER);
+            Color botClr = robot.getPixelColor(point.x, point.y + (SQUARE_HEIGHT/2) - PIXEL_BUFFER);
+
+            // close frame when complete
+            SwingUtilities.invokeAndWait(() -> {
+                frame.dispose();
```

Comment:
unless I am missing something you are creating a new JFrame for every call to createAndShowGUI() and over-writing the frame variable each time . so only the last one can be being disposed by this call, no ?

In fact you aren't even hiding them, so doesn't the screen get cluttered with a stack of them ?

Maybe you should only create the UI once and just update its L&F each time through.